src/state/swap/hooks.ts

```diff
@@ -11,12 +12,20 @@ import useParsedQueryString from '../../hooks/useParsedQueryString'
 import { isAddress } from '../../utils'
 import { AppDispatch, AppState } from '../index'
 import { useCurrencyBalances } from '../wallet/hooks'
-import { Field, replaceSwapState, selectCurrency, setRecipient, switchCurrencies, typeInput } from './actions'
+import { Field, replaceSwapState, selectCurrency, setRecipient, switchCurrencies, typeInput, setBanner } from './actions'
 import { SwapState } from './reducer'
 
 import { useUserSlippageTolerance } from '../user/hooks'
 import { computeSlippageAdjustedAmounts } from '../../utils/prices'
 
+export function useSwapBanner() {
+  const dispatch = useDispatch<AppDispatch>()
+  const banner = getBanner();
+  banner.then((ban) => {
+    dispatch(setBanner({ banner: ban }))
+  });
+}
```

Comment:
I believe this can be improved to be closer to the use we have on the state manager at the farms frontend:

https://github.com/ApeSwapFinance/apeswap-frontend/blob/staging/src/views/Lottery/hooks/useGetRecentLotteryRoundData.ts#L12

https://github.com/ApeSwapFinance/apeswap-frontend/blob/staging/src/state/strapi/useFetchPromoHome.ts#L10

These are good examples of how we use the state manager to load information brought from an API.